### PR TITLE
Inductor qlinear int8_bf16 with bmm

### PR DESCRIPTION
**Pull request by leslie-fang-intel**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #116604
* #116599
* #116937

**Summary**
Fix issue: https://github.com/pytorch/pytorch/issues/116492, `linear` will be decomposed into `bmm` when input dim exceeds 2 and not contiguous. Fix this issue by convert the pattern back into `qlinear`. This PR focus on int8_bf16 case following of https://github.com/pytorch/pytorch/pull/116599.

**Test Plan**
```
python -u -m pytest -s -v test_mkldnn_pattern_matcher.py -k test_qlinear_int8_mixed_bf16_input_dim_exceeds_2_and_not_contiguous
```

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

### CODE REVIEW

File: torch/_inductor/fx_passes/quantization.py [hunk 1]
```diff
@@ -1711,7 +1715,11 @@ def _generate_dequant_bmm_node_pattern(
         aten.bmm.default,
         CallFunction(
             aten.expand.default,
-            dequantize_per_tensor_activation_pattern,
+            _may_generate_pattern_with_dtype_convert(
+                dequantize_per_tensor_activation_pattern,
+                KeywordArg("autocast_act_dtype"),
+                dtype != torch.float32,
```
Comment (jgong5) [inline torch/_inductor/fx_passes/quantization.py:1721]: do we support both bf16 and fp16 or only bf16?

Comment (leslie-fang-intel) [reply torch/_inductor/fx_passes/quantization.py:1721]: I think we only support int8_bf16 now. For int8_fp16, it should be a new path and we need further enabling and verification.

Comment (jgong5) [reply torch/_inductor/fx_passes/quantization.py:1721]: Then shall we explicitly check `dtype == torch.bfloat16` here instead?

Comment (leslie-fang-intel) [reply torch/_inductor/fx_passes/quantization.py]: Thanks for the comment.

- Add the check of `assert dtype in [torch.float32, torch.bfloat16]`
- Change `dtype != torch.float32` to `dtype == torch.bfloat16`

Please help to take a look again.

**Review by jgong5 — APPROVED** [review]

**Pull request closed without merging**